test/e2e/test/generation/generation.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package generation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+)
+
+func getGeneration(obj client.Object, k *test.K8sClient) (int64, error) {
+	if err := k.Client.Get(context.Background(), types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()}, obj); err != nil {
+		return 0, err
+	}
+
+	return obj.GetGeneration(), nil
+}
+
+func getObservedGeneration(obj commonv1.HasObservedGeneration, k *test.K8sClient) (int64, error) {
+	if err := k.Client.Get(context.Background(), types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()}, obj); err != nil {
+		return 0, err
+	}
+
+	return obj.GetObservedGeneration(), nil
+}
+
+// RetrieveAgentGenerationsStep stores the current metadata.Generation, and status.ObservedGeneration into the given fields.
+func RetrieveAgentGenerationsStep(obj commonv1.HasObservedGeneration, k *test.K8sClient, generation, observedGeneration *int64) test.Step {
```

Comment:
Agent?

Reply:
I would expect these also to be used for the Elasticsearch tests but I think you based this branch of a version of main where the observedGeneration code for Elasticsearch does not exist yet. Worth merging main and using the same code for both?

Reply:
I removed agent from this in this PR.  Let me see about merging rebasing/merging main and fixing this in ES e2e tests also.

Reply:
Rebased main, and updated Elasticsearch e2e tests to use shared/common generation package.